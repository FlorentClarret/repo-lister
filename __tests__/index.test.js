/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

import MockDate from 'mockdate'

describe('With Enzyme', () => {
  it('App shows "List of posts" heading', () => {
    const app = shallow(<App posts={[{id: 1}]}/>)

    expect(app.find('h1').text()).toEqual('List of posts')
  })
})

describe('With Snapshot Testing', () => {
  it('App matches snapshot', () => {
    MockDate.set('1/1/2000')
    const component = renderer.create(<App posts={[{id: 2}]}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  afterEach(() => MockDate.reset())
})
