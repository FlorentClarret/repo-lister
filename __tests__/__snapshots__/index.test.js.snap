// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App matches snapshot 1`] = `
<main>
  <div>
    <header>
      <nav>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
         | 
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </nav>
    </header>
    <h1>
      List of posts
    </h1>
    <section>
      <article>
        <h2 />
        <p />
        <a
          href="/post/2"
          onClick={[Function]}
        >
          Read more...
        </a>
      </article>
    </section>
    <footer>
      <time
        data-dateepoch={320367600000}
        data-datetime="1980-02-25T23:00:00.000+00:00"
        title="February 25th 1980, 23:00:00"
      >
        almost 20 years ago
      </time>
      I'm here to stay
    </footer>
  </div>
</main>
`;
